web/client/plugins/widgetbuilder/CatalogServiceEditor.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { isEmpty } from 'lodash';
+
+import uuid from 'uuid';
+import CatalogServiceEditorComponent from '../../components/catalog/CatalogServiceEditor';
+import { DEFAULT_ALLOWED_PROVIDERS } from '../MetadataExplorer';
+
+const emptyService = {
+    url: "",
+    type: "wms",
+    title: "",
+    isNew: true,
+    showAdvancedSettings: false,
+    showTemplate: false,
+    hideThumbnail: false,
+    metadataTemplate: "<p>${description}</p>"
+};
+
+export default ({service: defaultService, catalogServices,
+    error = () => {}, onAddService = () => {}, isNew, dashboardServices, defaultServices, defaultSelectedService,
+    dashboardSelectedService, ...props}) => {
+    const [service, setService] = useState(isNew ? emptyService :
+        isEmpty(dashboardSelectedService) ? {...defaultServices[defaultSelectedService], old: defaultServices[defaultSelectedService]} :
+            {...dashboardServices[dashboardSelectedService], old: dashboardServices[dashboardSelectedService]} );
+
+
+    const existingServices = isEmpty(dashboardServices) ? defaultServices : dashboardServices;
+
+    const addNewService = () => {
+        if (!service.type !== 'tms' && !(!service.title || !service.url)) {
```

Comment:
```suggestion
        if (!service.type !== 'tms' && !(!service.title || !service.url)) { // TODO validation should be internal to catalog editor, in common.
```

Reply:
Okay not sure what this means @offtherailz